### ISSUE TITLE
Train a Toonify model  with glasses and hair style

### ISSUE THREAD

**Issue by onefish51** [eladrich/pixel2style2pixel#150]
Thanks for the amazing  work!  I've managed to train a toonify model on my own pair  dataset which is starting to look good. 
But I found the result without glasses, as the photo myself is with glasses ,the result is very bad !
![image](https://user-images.githubusercontent.com/21029719/121170439-cbc8b280-c887-11eb-8f04-6d7e0277956e.png)

I tested the pretrained models `psp_ffhq_toonify.pt` you given , the result is with  glasses 
![image](https://user-images.githubusercontent.com/21029719/121170251-886e4400-c887-11eb-8da4-110bd282fdc0.png)

so I load both to train  :
`
checkpoint_path_encoder = 'pretrained_models/psp_ffhq_encode.pt'

print('Loading pSp encoder from checkpoint: {}'.format(checkpoint_path_encoder))
print('Loading pSp decoder from checkpoint: {}'.format(self.opts.checkpoint_path))

ckpt_encoder = torch.load(checkpoint_path_encoder, map_location='cpu')
ckpt_decoder = torch.load(self.opts.checkpoint_path, map_location='cpu')
`

and at first output is with glasses ,but when I train more time output without glasses 
![image](https://user-images.githubusercontent.com/21029719/121171242-d5064f00-c888-11eb-8e72-12409fb5ef90.png)
![image](https://user-images.githubusercontent.com/21029719/121171365-00893980-c889-11eb-87fd-d6979caff3fc.png)

and I change  `id loss` to `moco loss` train ,the result is the same .

Can you give me some ideas ? thank you very much !

**Comment by yuval-alaluf** [comment]
If you have paired data, you may need to play around with the loss functions. The loss functions we used were tuned for training with no paired data. 
Also, was the generator used to create your paired data the same generator as the toonify generator we used?

**Comment by onefish51** [comment]
Thank you for your quickly reply !
I compare the checkpoint `pretrained_models/psp_ffhq_encode.pt` and the checkpoint I trained `best_model.pt` , the decoder parameter key is the same ,but the value is different . So I should Fixed decoder parameter when I train ?

**Comment by yuval-alaluf** [comment]
I am not sure that I fully follow. I am not sure you need the `psp_ffhq_encode` model.
If you have paired data, you can use a pre-trained decoder on your target toonify domain (which remains fixed throughout training). 
Then you train the network to translate real images to their toonified versions using supervised training. 
Am I missing something?

**Comment by onefish51** [comment]
According to my understanding, you meant that:

1. first I train `rosinality/stylegan2-pytorch` by my toons dataset to get toons StyleGAN2 generator;
2. second I train pspnet by the paired data and pre-trained StyleGAN2 generator(fixed pspnet decoder);
3. third I play around with the loss functions you provided when I train pspnet .

Am I missing something?

**Comment by onefish51** [comment]
besides in  #61 you said "Since StyleGAN2 doesn't have any batch norm layers we do not need to set eval(), even during training. Note, however, that by default the parameters of the generator are not added to the optimizer and therefore the generator remains fixed during training."
StyleGAN2 generator is fixed during pspnet training, in other word pspnet decoder parameters should be the same during the training . so I compare the decoder parameters  between `iteration_35000.pt` and `iteration_5000.pt`, I found yes ! yesterday  I compare  the decoder parameters  between `psp_ffhq_encode.pt` and  my toons checkpoint `best_model.pt`, it's my wrong !

**Comment by yuval-alaluf** [comment]
Everything you said above is correct. 
However, regarding why `psp_ffhq_encode.pt` and `best_model.pt` have different decoder parameters, this makes sense. The psp_ffhq_encode.pt checkpoint uses a decoder trained on FFHQ to output real face images. Your decoder is trained on toon images to output the toonify images. Therefore I makes sense the two are different. 
The weights of the best_model.pt should be identical to the weights in iteration_5000.pt, for example.

**Comment by onefish51** [comment]
yes, you are right!
